hazelcast/src/main/java/com/hazelcast/query/impl/UnorderedIndexStore.java

```diff
@@ -148,62 +151,68 @@ public void clear() {
         }
     }
 
+    @SuppressWarnings({"unchecked", "checkstyle:npathcomplexity"})
     @Override
-    public Set<QueryableEntry> getRecords(Comparable value) {
-        takeReadLock();
-        try {
-            if (value instanceof IndexImpl.NullObject) {
-                return toSingleResultSet(recordsWithNullValue);
-            } else {
-                return toSingleResultSet(recordMap.get(value));
-            }
-        } finally {
-            releaseReadLock();
-        }
-    }
-
-    @Override
-    public Set<QueryableEntry> getRecords(Set<Comparable> values) {
+    public Set<QueryableEntry> getRecords(Comparable from, boolean fromInclusive, Comparable to, boolean toInclusive) {
         takeReadLock();
         try {
             MultiResultSet results = createMultiResultSet();
-            for (Comparable value : values) {
-                Map<Data, QueryableEntry> records;
-                if (value instanceof IndexImpl.NullObject) {
-                    records = recordsWithNullValue;
-                } else {
-                    records = recordMap.get(value);
+            if (from.compareTo(to) == 0) {
+                if (!fromInclusive || !toInclusive) {
+                    return results;
                 }
+
+                Map<Data, QueryableEntry> records = recordMap.get(from);
                 if (records != null) {
                     copyToMultiResultSet(results, records);
                 }
+                return results;
+            }
+
+            int fromBound = fromInclusive ? 0 : +1;
+            int toBound = toInclusive ? 0 : -1;
+            for (Map.Entry<Comparable, Map<Data, QueryableEntry>> recordMapEntry : recordMap.entrySet()) {
+                Comparable value = recordMapEntry.getKey();
+                if (value.compareTo(from) >= fromBound && value.compareTo(to) <= toBound) {
+                    Map<Data, QueryableEntry> records = recordMapEntry.getValue();
+                    if (records != null) {
+                        copyToMultiResultSet(results, records);
+                    }
+                }
             }
             return results;
         } finally {
             releaseReadLock();
         }
     }
 
+    @Override
+    public String toString() {
+        return "UnorderedIndexStore{" + "recordMap=" + recordMap.size() + '}';
```

Comment:
?"recordMap size="

Reply:
Removed `toString` for index stores completely, it doesn't look like it was that much useful.